src/parser.cpp

```diff
@@ -1074,9 +1074,15 @@ ComponentPtr Parser::ParserImpl::loadComponentRef(const ModelPtr &model, const X
             mParser->addIssue(issue);
         }
 
-        if ((parentComponent) && (childComponent)) {
+        if (childComponent) {
             // Set parent/child encapsulation relationship.
-            parentComponent->addComponent(childComponent);
+            if (parentComponent) {
```

Comment:
Should read: `if (parentComponent != nullptr) {`.

Reply:
OK ... then I'll also change the line above which reads `if(childComponent) {` too ... !

Reply:
Actually ... this is the idiom throughout the whole of the parser :(

Reply:
Ok, forget about it then. Will have to be done as part of our big cleaning up.